orthoinference/Jenkinsfile

```diff
@@ -0,0 +1,229 @@
+import groovy.json.JsonSlurper
+
+pipeline{
+	agent any
+	
+	stages{
+		stage('Check if upstream builds succeeded'){
+			steps{
+				script{
+					def orthopairsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/Orthopairs/lastBuild/api/json"
+					def orthopairsStatusJson = new JsonSlurper().parseText(orthopairsStatusUrl.getContent())
+					if(orthopairsStatusJson['result'] != "SUCCESS"){
+						error("Most recent Orthopairs build status: " + orthopairsStatusJson['result'])
+					}
+			    	}
+				script{
+					def updateStIdsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/UpdateStableIdentifiers/lastBuild/api/json"
+					def updateStIdsStatusJson = new JsonSlurper().parseText(updateStIdsStatusUrl.getContent())
+					if(updateStIdsStatusJson['result'] != "SUCCESS"){
+						error("Most recent UpdateStableIdentifiers build status: " + updateStIdsStatusJson['result'])
+					}
+				}	
+		    	}
+	    	}
+		stage('Setup: Backup release_current'){
+			steps{
+				script{
+					dir('orthoinference'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
```

Comment:
A similar question about the password on the command-line to the AddLinks Jenkins PR.  Is it okay to do and is it necessary/doable to provide it in a different way?

Reply:
I had this reply in the addlinks version of this PR:

> I am sure its doable, but this is probably more easily tackled when we actually deploy this on the release server, in case there are any differences in how the Jenkins user is utilized there than on my computer. For now, Jenkins is very good for making sure credentialed values show up as asterisks. Definitely can consider the use of the `my.cnf` file once its in place, though I don't know if having to switch between the local release server and curator server are going to make things difficult, _a la_ the `pass2production.sh` issue we had for release 70

Reply:
Sounds good - we can work on it in a new PR after trying it on a server.